guacamole-docker/bin/start.sh

```diff
@@ -594,6 +594,46 @@ END
 
 }
 
+## Adds properties to guacamole.properties which select the SAML
+## authentication provider, and configure it to connect to the specified SAML
+## provider.
+##
+associate_saml() {
+
+    # Verify required parameters are present
+    if [ -z "$SAML_IDP_METADATA_URL" ]
+    then
+        cat <<END
+FATAL: Missing required environment variables
+-------------------------------------------------------------------------------
+If using an openid authentication, you must provide each of the following
+environment variables:
+
+    SAML_IDP_METADATA_URL           The URI of the XML metadata file that from the SAML Identity
+                                    Provider
+END
+        exit 1;
+    fi
+
+    # Update config file
+    set_property          "saml-idp-metadata-url"            "$SAML_IDP_METADATA_URL"
```

Comment:
I think this has to be optional, as well, unfortunately. Only the SAML support itself is going to be able to accurately sanity check these variables.

It's possible for the `saml-idp-metadata-url` property to be omitted, and the required values instead provided through individual properties (like `saml-entity-id` and `saml-callback-url`), and it's also possible to specify `saml-idp-metadata-url` _and_ individual properties to make up for any information not included in the XML pointed to by that URL.